File: src/main/java/site/gachontable/gachontablebe/domain/auth/domain/KakaoProfile.java
package site.gachontable.gachontablebe.domain.auth.domain;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;

public record KakaoProfile(String username, String tel) {

    public static KakaoProfile from(String jsonResponseBody) {
        JsonObject object = JsonParser.parseString(jsonResponseBody).getAsJsonObject();

        JsonObject kakaoAccount = object.getAsJsonObject("kakao_account");
        String username = kakaoAccount.get("name").getAsString();
        String tel = kakaoAccount.get("phone_number").getAsString();

        return new KakaoProfile(username, tel);
    }
}

File: src/main/java/site/gachontable/domain/auth/domain/KakaoProfile.kt
package site.gachontable.domain.auth.domain

import com.google.gson.JsonParser

data class KakaoProfile(
    val username: String,
    val tel: String,
) {
    companion object {
        fun from(jsonResponseBody: String): KakaoProfile {
            val `object` = JsonParser.parseString(jsonResponseBody).getAsJsonObject()

            val kakaoAccount = `object`.getAsJsonObject("kakao_account")
            val username = kakaoAccount.get("name").asString
            val tel = kakaoAccount.get("phone_number").asString

            return KakaoProfile(username, tel)
        }
    }
}
